app/src/components/Table/Table.jsx

```diff
@@ -1,15 +1,61 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { capitalize } from '../../utils';
 import './Table.scss';
 
+const SORT = {
+  DEFAULT: 0,
+  ASCENDING: 1,
+  DESCENDING: 2
+};
+
+const initSort = { type: SORT.DEFAULT, value: 'default', by: '' };
+
 function getValue(row, key, get, transform) {
   const value = get ? get(row) : row[key];
   return [transform ? transform(value, row) : value, value];
 }
 
 function Table({ rows, columns, highlightedIndex, onRowClicked, clickable }) {
+  const [sortData, setSortData] = useState(initSort);
+
+  const handleClick = key => {
+    let sortNext = SORT.DEFAULT;
+    if (sortData.type === SORT.DEFAULT) {
+      sortNext = SORT.ASCENDING;
+    } else if (sortData.type === SORT.ASCENDING) {
+      sortNext = SORT.DESCENDING;
+    }
+
+    const data = {
+      type: sortNext,
+      value: Object.keys(SORT)
+        .find(val => SORT[val] === sortNext)
+        .toLowerCase(),
+      by: key
+    };
+
+    setSortData(data);
+  };
+
+  const handleSort = (a, b) => {
+    if (sortData.type === SORT.ASCENDING) {
+      return b[sortData.by] - a[sortData.by];
+    }
+    if (sortData.type === SORT.DESCENDING) {
+      return a[sortData.by] - b[sortData.by];
+    }
+
+    return 0;
+  };
+
+  useEffect(() => {
```

Comment:
What is this for? Is it for restoring the default sorting when the table gets unmounted?

Reply:
Yeah, otherwise if you sort by example by 'rank' in skilling, then change the tab to bossing, 'rank' will be sorted by default. Not sure if that's a bad thing though.

Reply:
ah makes sense